src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,29 @@
 package mate.academy;
 
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private List<EventListener> listeners = new CopyOnWriteArrayList<>();
+    private ExecutorService executor = Executors.newCachedThreadPool();
 
+    public void registerListener(EventListener listener) {
+        listeners.add(listener);
     }
 
     public void deregisterListener(EventListener listener) {
-
+        listeners.remove(listener);
     }
 
     public void notifyEvent(Event event) {
-
+        for (EventListener listener : listeners) {
+            executor.submit(() -> listener.onEvent(event));
+        }
```

Comment:
Submitting tasks to an executor inside a loop can be inefficient and may lead to out-of-order execution. Consider refactoring to use invokeAll or another mechanism that can handle a collection of tasks more efficiently.